poseidon/controllers/faucet/parser.py

```diff
@@ -301,7 +301,7 @@ def config(self, config_file, action, port, switch, rules_file=None, endpoints=N
                                                         match = True
                                 if match:
                                     matches += 1
-                            elif r['rule']['device_key'] == 'behavior':
+                            elif r['rule'] and r['rule']['device_key'] == 'behavior':
```

Comment:
these need to check if `device_key` exists, since there is the existence of a rule with only `acls`